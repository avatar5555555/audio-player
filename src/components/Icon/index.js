import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const icons = {
  play: require('./icons/Play').default,
  pause: require('./icons/Pause').default,
  forward: require('./icons/Forward').default,
  backward: require('./icons/Backward').default,
  volume: require('./icons/Volume').default,
  cross: require('./icons/Cross').default,
  cd: require('./icons/Cd').default,
  stop: require('./icons/Stop').default
}

const IconRoot = styled.div`
  width: ${p => (p.small ? '15px' : '100px')};
  height: ${p => (p.small ? '15px' : '100px')};
`
const Icon = ({ name, small }) => {
  const Component = icons[name]

  return name ? (
    <IconRoot small={small}>
      <Component className="d-flex" />
    </IconRoot>
  ) : null
}

Icon.propTypes = {
  name: PropTypes.string,
  small: PropTypes.bool
}

Icon.defaultProps = {
  name: '',
  small: true
}

export default Icon
