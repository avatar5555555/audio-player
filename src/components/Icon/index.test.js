import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import Icon, { icons } from './index'

const Images = Object.keys(icons)

describe('Icon', () => {
  it('renders all icons', () => {
    Images.forEach(icon => shallow(<Icon name={icon} />))
  })

  it('match all icons snapshots', () => {
    Images.forEach(icon =>
      expect(toJSON(shallow(<Icon name={icon} />))).toMatchSnapshot()
    )
  })

  it('match null snapshot', () => {
    expect(toJSON(shallow(<Icon />))).toMatchSnapshot()
  })
})
