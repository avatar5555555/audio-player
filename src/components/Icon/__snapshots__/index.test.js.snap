// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon match all icons snapshots 1`] = `
<styled.div
  small={true}
>
  <Play
    className="d-flex"
  />
</styled.div>
`;

exports[`Icon match all icons snapshots 2`] = `
<styled.div
  small={true}
>
  <Pause
    className="d-flex"
  />
</styled.div>
`;

exports[`Icon match all icons snapshots 3`] = `
<styled.div
  small={true}
>
  <Forward
    className="d-flex"
  />
</styled.div>
`;

exports[`Icon match all icons snapshots 4`] = `
<styled.div
  small={true}
>
  <Backward
    className="d-flex"
  />
</styled.div>
`;

exports[`Icon match all icons snapshots 5`] = `
<styled.div
  small={true}
>
  <Volume
    className="d-flex"
  />
</styled.div>
`;

exports[`Icon match all icons snapshots 6`] = `
<styled.div
  small={true}
>
  <Cross
    className="d-flex"
  />
</styled.div>
`;

exports[`Icon match all icons snapshots 7`] = `
<styled.div
  small={true}
>
  <Cd
    className="d-flex"
  />
</styled.div>
`;

exports[`Icon match all icons snapshots 8`] = `
<styled.div
  small={true}
>
  <Stop
    className="d-flex"
  />
</styled.div>
`;

exports[`Icon match null snapshot 1`] = `""`;
